src/Framework/Analyzer/test/MinimalActions/DisallowReturningActionResultsFromMapMethodsTest.cs

```diff
@@ -0,0 +1,255 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Analyzer.Testing;
+
+namespace Microsoft.AspNetCore.Analyzers.MinimalActions;
+
+public partial class DisallowReturningActionResultsFromMapMethodsTest
+{
+    private TestDiagnosticAnalyzerRunner Runner { get; } = new(new MinimalActionAnalyzer());
+
+    [Fact]
+    public async Task MinimalAction_ReturningIResult_Works()
+    {
+        // Arrange
+        var source = @"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+var webApp = WebApplication.Create();
+webApp.MapGet(""/"", () => Results.Ok(""Hello""));
+";
+        // Act
+        var diagnostics = await Runner.GetDiagnosticsAsync(source);
+
+        // Assert
+        Assert.Empty(diagnostics);
+    }
+
+    [Fact]
+    public async Task MinimalAction_ReturningCustomIResult_Works()
+    {
+        // Arrange
+        var source = @"
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+var webApp = WebApplication.Create();
+webApp.MapGet(""/"", () => new CustomResult());
+
+class CustomResult : IResult
+{
+    public Task ExecuteAsync(HttpContext context) => Task.CompletedTask;
+}
+";
+        // Act
+        var diagnostics = await Runner.GetDiagnosticsAsync(source);
+
+        // Assert
+        Assert.Empty(diagnostics);
+    }
+
+    [Fact]
+    public async Task MinimalAction_ReturningIResultConditionallyWorks()
+    {
+        // Arrange
+        var source = @"
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+var webApp = WebApplication.Create();
+webApp.MapGet(""/"", (int id) =>
+{
+    if (id == 0)
+    {
+        return Results.NotFound();
+    }
+
+    return Results.Ok(""Here you go"");
+});
+
+";
+        // Act
+        var diagnostics = await Runner.GetDiagnosticsAsync(source);
+
+        // Assert
+        Assert.Empty(diagnostics);
+    }
+
+    [Fact]
+    public async Task MinimalAction_ReturningTypeThatImplementsIResultAndActionResultFails()
```

Comment:
Good test

Reply:
Does this work or fail?
From the code it looks like it should work (no diagnostics) but the method name suggests otherwise.
```suggestion
    public async Task MinimalAction_ReturningTypeThatImplementsIResultAndActionResultWorks()
```

Reply:
```suggestion
    public async Task MinimalAction_ReturningTypeThatImplementsIResultAndActionResultDoesNotProduceDiagnostics()
```